vignettes/data-transform-as-shiny-module.Rmd

```diff
@@ -160,6 +160,70 @@ if (interactive()) {
 When a custom transformation is used, the UI for the transformation is placed below the filter panel.
 However, there is a way to customize the placement of the UI inside the module content.
 
-This can only be done when the module allocates a dedicated space for the transform UI.
+In order to place the transformation UI inside the module there are few things one has to do:
+1. Create a custom module wrapper function.
+2. Call the desired module in the module wrapper function and store it in a variable so it can be modified.
+3. Modify the UI of the module with the transform UI at the desired location by calling the `ui_transform_data`. Note that in order for the transform to work you need to change the name space of the id by passing `NS(gsub("-module$", "", id), "data_transform")`.
+4. Set the `custom_ui` attribute of the `module$transformers` to `TRUE`.
+
+Now the custom module should embed the transformation UI inside the module content.
+
+Here is an example of a custom module wrapper function that modifies the `example_module` module.
+```{r}
+example_module_encoding <- function(label = "example module (on encoding)", datanames = "all", transformers = list()) {
+  mod <- example_module(label, datanames, transformers)
+  mod$ui <- function(id) {
+    ns <- NS(id)
+    teal.widgets::standard_layout(
+      output = verbatimTextOutput(ns("text")),
+      encoding = tags$div(
+        ui_transform_data(NS(gsub("-module$", "", id), "data_transform"), transformers),
+        selectInput(ns("dataname"), "Choose a dataset", choices = NULL),
+        teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code")
+      )
+    )
```

Comment:
Maybe we can invent an example that is not based on `example_module`?
Maybe you can create a simple `module` that uses `ui` and `server` and you can show an application on whether it uses the default placement, and a custom placement?

This way it might be a bit more clear, instead of showing an example where we use pre-defined module and then we adjust it's UI. Could be simpler to have simple module with ui and server and then version adjusted with ui_transform_data added

Reply:
I wanted to do it as you mentioned by creating a custom module and then injecting this.
However, The typical use case for this would be to inject the custom transform in a module from a package like tmc. And, the vignette would become longer with custom module code which might distract the users from the actual important code needed to position the transform.
So, I decided to just place the transform to the custom location from the example mentioned above.

Reply:
hmmm, I'll think about it

Reply:
I think it is fine to use `example_module` if we got with the wrapper that I proposed